nats-concepts/jetstream/consumers.md

```diff
@@ -72,7 +72,13 @@ When first consuming messages, start with messages on or after this time. The co
 
 ## DeliverySubject
 
-The subject to deliver observed messages. Not allowed for pull subscriptions. A delivery subject is required for queue subscribing as it configures a subject that all the queue consumers should listen on.
+The delivery subject is used by NATS to orchestrate how messages reach subscriptions. When a message is published to a JetStream, NATS will know of all consumers that have registered an interested in that message. It will then deliver the message to each consumer via their `DeliverySubject`. In this sense, a `DeliverySubject` can be considered like an "inbox" for each subscription.
+
+For this reason, `DeliverySubject` is not allowed for pull subscriptions, since in that case the subscription is fetching the messages for itself. Similarly, a delivery subject is required for all push subscriptions, as well as for queue subscribing as it configures a subject that all the queue consumers should listen on.
+
+The delivery subject should not be confused with the stream's subject. In fact, if a delivery subject had the same value as a stream's subject, it could create an infinite loop as NATS would re-deliver messages to the same subjects again and again. NATS will prevent the creation of this by emitting the following error: `Consumer creation failed: consumer deliver subject forms a cycle (10081)`.
```

Comment:
```suggestion
The deliver subject should not be confused with the stream's subject. In fact, if a deliver subject had the same value as a stream's subject, NATS will emit the following error: `Consumer creation failed: consumer deliver subject forms a cycle (10081)` to prevent published messages from being delivered on an overlapping subject.
```